desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -97,25 +105,65 @@ def handle_on_link_shared(channel_id, message_ts, links):
     try:
       if path == '/hue/editor' and id_type == 'editor':
         doc = Document2.objects.get(id=qid)
+        doc_type = 'Editor'
       elif path == '/hue/gist' and id_type == 'uuid' and ENABLE_GIST_PREVIEW.get():
         doc = _get_gist_document(uuid=qid)
+        doc_type = 'Gist'
       else:
         raise PopupException(_("Cannot unfurl link"))
     except Document2.DoesNotExist:
       msg = "Document with {key}={value} does not exist".format(key='uuid' if id_type == 'uuid' else 'id', value=qid)
       raise PopupException(_(msg))
 
-    doc_data = json.loads(doc.data)
-    statement = doc_data['snippets'][0]['statement_raw'] if id_type == 'editor' else doc_data['statement_raw']
-    dialect = doc_data['dialect'].capitalize() if id_type == 'editor' else doc.extra.capitalize()
-    created_by = doc.owner.get_full_name() or doc.owner.username
-    
-    payload = _make_unfurl_payload(item['url'], statement, dialect, created_by)
+    payload = _make_unfurl_payload(item['url'], id_type, doc, doc_type)
     response = slack_client.chat_unfurl(channel=channel_id, ts=message_ts, unfurls=payload)
     if not response['ok']:
       raise PopupException(_("Cannot unfurl link"), detail=response["error"])
 
-def _make_unfurl_payload(url, statement, dialect, created_by):
+
+def query_result(request, notebook):
+  snippet = notebook['snippets'][0]
+  snippet['statement'] = notebook['snippets'][0]['statement_raw']
+
+  query_execute = _execute_notebook(request, notebook, snippet)
+
+  history_uuid = query_execute['history_uuid']
+  status = _check_status(request, operation_id=history_uuid)
+  if status['query_status']['status'] == 'available':
+      response = _fetch_result_data(request, operation_id=history_uuid)
+
+  return response['result']
+
+def _make_result_table(result):
+  meta = []
+  for field in result['meta']:
+    meta.append(field['name'])
+
+  table = PrettyTable()
+  table.field_names = meta
+  table.add_rows(result['data'])
+  return table
+
+def _make_unfurl_payload(url, id_type, doc, doc_type):
+  doc_data = json.loads(doc.data)
+  statement = doc_data['snippets'][0]['statement_raw'] if id_type == 'editor' else doc_data['statement_raw']
+  dialect = doc_data['dialect'].capitalize() if id_type == 'editor' else doc.extra.capitalize()
+  created_by = doc.owner.get_full_name() or doc.owner.username
+
+  # Mock request for query execution and fetch result
+  user = rewrite_user(User.objects.get(username=doc.owner.username))
+  request = MockRequest(user=user)
+
+  result = 'Result is not available for Gist.'
+  if id_type == 'editor':
+    try:
```

Comment:
Curious why we would need the try/except?

Reply:
To catch the QueryExpired exception raised by _check_status @romainr

Reply:
So cleaner to `except QueryExpired:`

Reply:
`@api_error_handler` decorator has multiple exception types which can be raised (along with check status we have fetch_result and execute in that try/except too)
So how about using ` except Exception:` instead of `except QueryExpired:` @romainr or something better?